developer/src/keyman-vscode-plugin/src/extensionCallbacks.mts

```diff
@@ -0,0 +1,266 @@
+/*
+ * Keyman is copyright (C) SIL Global. MIT License.
+ *
+ * Code related to Extension support for callbacks.
+ */
+
+import { CompilerCallbackOptions, CompilerCallbacks, CompilerError, CompilerEvent, CompilerFileCallbacks, CompilerFileSystemAsyncCallbacks, CompilerFileSystemCallbacks, CompilerNetAsyncCallbacks, CompilerPathCallbacks, FileSystemFolderEntry } from "@keymanapp/developer-utils";
+import * as fs from 'fs';
+import * as path from 'node:path';
+
+
+interface ErrorWithCode {
+    code?: string;
+}
+
+function resolveFilename(baseFilename: string, filename: string) {
+  const basePath =
+    baseFilename.endsWith('/') || baseFilename.endsWith('\\') ?
+    baseFilename :
+    path.dirname(baseFilename);
+  // Transform separators to platform separators -- we are agnostic
+  // in our use here but path prefers files may use
+  // either / or \, although older kps files were always \.
+  if(path.sep == '/') {
+    filename = filename.replace(/\\/g, '/');
+  } else {
+    filename = filename.replace(/\//g, '\\');
+  }
+  if(!path.isAbsolute(filename)) {
+    filename = path.resolve(basePath, filename);
+  }
+  return filename;
+}
+
+class NodeCompilerFileSystemAsyncCallbacks implements CompilerFileSystemAsyncCallbacks {
+  async exists(filename: string): Promise<boolean> {
+    return fs.existsSync(filename);
+  }
+
+  async readFile(filename: string): Promise<Uint8Array> {
+    return fs.readFileSync(filename);
+  }
+
+  async readdir(filename: string): Promise<FileSystemFolderEntry[]> {
+    return fs.readdirSync(filename).map(item => ({
+      filename: item,
+      type: fs.statSync(path.join(filename, item))?.isDirectory() ? 'dir' : 'file'
+    }));
+  }
+
+  resolveFilename(baseFilename: string, filename: string): string {
+    return resolveFilename(baseFilename, filename);
+  }
+}
+
+class NodeCompilerNetAsyncCallbacks implements CompilerNetAsyncCallbacks {
+  async fetchBlob(url: string, options?: RequestInit): Promise<Uint8Array> {
+    try {
+      const response = await fetch(url, options);
+      if(!response.ok) {
+        throw new Error(`HTTP error ${response.status}: ${response.statusText}`);
+      }
+
+      const data = await response.blob();
+      return new Uint8Array(await data.arrayBuffer());
+    } catch(e) {
+      throw new Error(`Error downloading ${url}`, {cause:e});
+    }
+  }
+
+  async fetchJSON(url: string, options?: RequestInit): Promise<any> {
+    try {
+      const response = await fetch(url, options);
+      if(!response.ok) {
+        throw new Error(`HTTP error ${response.status}: ${response.statusText}`);
+      }
+
+      return await response.json();
+    } catch(e) {
+      throw new Error(`Error downloading ${url}`, {cause:e});
+    }
+  }
+}
+
+
+export class ExtensionCallbacks implements CompilerCallbacks {
```

Comment:
the output path is different, via a callback instead of console.out 

- [ ] try to refactor the output section

perhaps a MessageHandler object or something.